### PR TITLE
EmbeddedAtomModel

### PR DESCRIPTION
**Pull request by grayseff**
Implementation of the Embedded Atom Model. Major changes include:

- EnergyEvaluation.jl:  Functions related to the calculation of the embedded atom potentials.

-  MCRun.jl: Includes a new swap_config for use with embedded atom components, along with acc_test and mc_step.

Future implementation includes a proper new_dist2_vec and new_component_vec to prevent memory leakage.

### CODE REVIEW

File: scripts/run_Cu78.jl
Comment (ElkePahl) [inline scripts/run_Cu78.jl:181]: It would be good to have the formula here/ meaning of parameters, perhaps reference to paper(s) where you have taken parameters from.

File: scripts/run_Cu78.jl
Comment (ElkePahl) [inline scripts/run_Cu78.jl:1]: It would be good if name of script makes it clear that it is for EAM?

File: src/EnergyEvaluation.jl
Comment (ElkePahl) [inline src/EnergyEvaluation.jl:436]: I can't find any function calc_all_components?
All calc_components methods should be described in one docu string.

File: src/EnergyEvaluation.jl
Comment (ElkePahl) [inline src/EnergyEvaluation.jl:402]: add formula to docu string

File: src/MCRun.jl
Comment (ElkePahl) [inline src/MCRun.jl:132]: CAn you remind me what is this about?

**Review by ElkePahl — COMMENTED** [review]
I don't approve yet, as I am not sure if missing function calc_all_components is essential

File: src/MCRun.jl
Comment (grayseff) [reply src/MCRun.jl:132]: There is a vector of parameters rho and phi representing the two-body and density contributions of each atom, we need to pass the vector of __new__ parameters to the acc_test function because we need to exchange them for the old ones if the move is accepted.

File: src/EnergyEvaluation.jl
Comment (grayseff) [reply src/EnergyEvaluation.jl:436]: This is a leftover from an earlier version where the two methods were separate functions, now corrected

File: scripts/run_Cu78.jl
Comment (grayseff) [reply scripts/run_Cu78.jl:1]: This works for RuNNer or the EAM as both potentials are defined.

File: scripts/run_Cu78.jl
Comment (grayseff) [reply scripts/run_Cu78.jl:181]: Fixing this now

File: src/MCRun.jl
Comment (ElkePahl) [reply src/MCRun.jl:132]: All seems fine then? I guess you still have to commit the changes?

**Review by ElkePahl — APPROVED** [review]
all fine

**Pull request merged**
